task6

csn18 opened the pull request:
(main.js) Add function ajaxSearch
(html) Create new file task6.html
(server.py) Add app.route(/task6), add app.route(/task6search)

csn18 committed:
(main.js) Add function ajaxSearch
(html) Create new file task6.html
(server.py) Add app.route(/task6), add app.route(/task6search)

Doaxan commented on server.py:
```diff
@@ -1,4 +1,4 @@
-from flask import Flask, render_template, request
+from flask import Flask, render_template, request, jsonify, json
```
json?

Doaxan commented on static/js/main.js:
```diff
@@ -25,16 +26,48 @@ function ajaxPagination() {
+        data: {text: textInSearchBox},
+        success: (result) => {
+          countryId = result['countryId']
+          console.log(countryId)
```
?

Doaxan commented on static/js/main.js:
```diff
@@ -25,16 +26,48 @@ function ajaxPagination() {
+
+      $.ajax({
+        url: '/task6search',
+        type: 'POST',
```
Please use GET request, for **get** data

Doaxan commented on static/js/main.js:
```diff
@@ -25,16 +26,48 @@ function ajaxPagination() {
+        success: (result) => {
+          countryId = result['countryId']
+          console.log(countryId)
+          res = result['cities']
```
?

Doaxan replied:
What is "res"?

Doaxan reviewed (commented):
Please resolve duplicates, follow https://github.com/csn18/cities_app/issues/2

csn18 committed:
(main.js) Fix code (40 line)
(server.py) Fix duplicate, Fix import lines (PEP8)

csn18 committed:
(server.py) Changed data retrieval method

csn18 committed:
(server.py) Fix duplicate pagination

Doaxan commented on server.py:
```diff
@@ -1,5 +1,6 @@
-from flask import Flask, render_template, request, jsonify, json
-import mysql.connector, math, re
+import math
```
Maybe use one import, instead of several?

Doaxan replied:
https://www.python.org/dev/peps/pep-0008/#imports

Doaxan commented on server.py:
```diff
@@ -40,28 +41,7 @@ def index3():
-                               )
-
-    return render_template('main/task3.html', id_countries=id_countries_db)
+    return render_page('main/task3.html', countries_id_query, id_countries_db)
```
Instead of use full path, just get current path from request.

Doaxan reviewed (changes requested):
Not all the comments have been fixed, including the duplicates. Please take care of them.

csn18 committed:
(main.js) Fix naming variable (40 line)
(server.py) Fix duplicate(query)
Added getting path from app.route
Added function "request_response" for query

csn18 committed:
(server.py) Fix duplication
